ta4j-core/src/main/java/org/ta4j/core/indicators/ParabolicSarIndicator.java

```diff
@@ -126,7 +126,10 @@ protected Num calculate(int index) {
                 currentExtremePoint = getBarSeries().getBar(index).getLowPrice(); // put point on max
                 minMaxExtremePoint = currentExtremePoint;
             } else { // up trend is going on
-                currentExtremePoint = new HighestValueIndicator(highPriceIndicator, index - startTrendIndex)
+                Num lowestPriceOfTwoPreviousBars = new LowestValueIndicator(lowPriceIndicator, Math.min(2, index - startTrendIndex)).getValue(index - 1);
```

Comment:
What is the reason for this change? 
current:  Uptrend or downtrend means we have to calculate the extreme point. 
yours: You are replacing this with calculating the lowest min price of the trend or at least of the last two bars if a up trend has been detected?

Reply:
You may ask this question from who invented this indicator. I used your library to not overwrite what is written before and tested some of my strategies on it and then found out that the ParabolicSarIndicator calculates wrong, so I searched and then found a Wikipedia page for it and do what it says and then tested it on TradingView values, then it seems that works great. 

check it out in the link below.

_**If the next period's SAR value is inside (or beyond) the current period or the previous period's price range, the SAR must be set to the closest price bound. For example, if in an upward trend, the new SAR value is calculated and if it results to be more than today's or yesterday's lowest price, it must be set equal to that lower boundary.**_

https://en.wikipedia.org/wiki/Parabolic_SAR

Thanks for your consideration.